src/Log/NullSemanticLogger.php

```
@@ -0,0 +1,45 @@
+ * No-op semantic logger
+ *
+ * Zero-cost default when cache logging is turned off. open() returns an empty
+ * id (which SafeSemanticLogger and disciplined call sites treat as "no close
+ * needed"), and flush() returns an empty log session.
+ */
+final class NullSemanticLogger implements SemanticLoggerInterface
+{
+    private const EMPTY_SCHEMA_URL = 'https://koriym.github.io/Koriym.SemanticLogger/schemas/semantic-log.json';
+
+    #[Override]
+    public function open(AbstractContext $context): string
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$context'. (undefined)

(UnusedFormalParameter)